addons/stock_account/models/product.py

```diff
@@ -338,8 +339,10 @@ def _run_fifo(self, quantity, company):
 
             if float_is_zero(qty_to_take_on_candidates, precision_rounding=self.uom_id.rounding):
                 if float_is_zero(candidate.remaining_qty, precision_rounding=self.uom_id.rounding):
-                    next_candidates = candidates.filtered(lambda svl: svl.remaining_qty > 0)
-                    new_standard_price = next_candidates and next_candidates[0].unit_cost or new_standard_price
+                    next_candidate = candidates.filtered(lambda svl: svl.remaining_qty > 0)[0:]
```

Comment:
```suggestion
    next_candidate = candidates.filtered(lambda svl: svl.remaining_qty > 0)[:1]
```
you are getting the whole list (from index 0 to infinite), you probably want `[:1]` to only get the first element (or 0 element if the list is empty)